Tests/UnitTests/Core/Material/GridSurfaceMaterialTests.cpp

```diff
@@ -223,37 +529,70 @@ BOOST_AUTO_TEST_CASE(GridIndexedMaterial2D) {
   eqeqGrid.atPosition(Point{0.5, std::numbers::pi * 0.75}) = 0u;   // vacuum
 
   // With radius 20
-  auto boundToGrid = std::make_unique<const LocalToZPhi>(20.);
-  Acts::IndexedSurfaceMaterial<EqEqGrid>::BoundToGridLocalDelegate bToZPhi;
-  bToZPhi.connect<&LocalToZPhi::l2ZPhi>(std::move(boundToGrid));
+  auto boundToGridT1 = std::make_unique<const LocalToZPhi>(20.);
+  Acts::IndexedSurfaceMaterial<EqEqGrid>::BoundToGridLocalDelegate bToZPhiT1;
+  bToZPhiT1.connect<&LocalToZPhi::l2ZPhi>(std::move(boundToGridT1));
 
   // With z shift 10
-  auto globalToGrid = std::make_unique<const GlobalToZPhi>(10.);
-  Acts::IndexedSurfaceMaterial<EqEqGrid>::GlobalToGridLocalDelegate gToZphi;
-  gToZphi.connect<&GlobalToZPhi::g2ZPhi>(std::move(globalToGrid));
+  auto globalToGridT1 = std::make_unique<const GlobalToZPhi>(10.);
+  Acts::IndexedSurfaceMaterial<EqEqGrid>::GlobalToGridLocalDelegate gToZphiT1;
+  gToZphiT1.connect<&GlobalToZPhi::g2ZPhi>(std::move(globalToGridT1));
 
   // Create the indexed material grid
   Acts::IndexedSurfaceMaterial<EqEqGrid> ism(
-      std::move(eqeqGrid), Acts::IndexedMaterialAccessor{std::move(material)},
-      std::move(bToZPhi), std::move(gToZphi));
+      std::move(eqeqGrid), Acts::IndexedMaterialAccessor{std::move(materialT1)},
+      std::move(bToZPhiT1), std::move(gToZphiT1));
+
+  // Test with proto grid greation
+  auto materialT2 = material;
+
+  auto boundToGridT2 = std::make_unique<const LocalToZPhi>(20.);
+  Acts::GridAccess::BoundToGridLocal2DimDelegate bToZPhiT2;
+  bToZPhiT2.connect<&LocalToZPhi::l2ZPhi>(std::move(boundToGridT2));
+
+  auto globalToGridT2 = std::make_unique<const GlobalToZPhi>(10.);
+  Acts::GridAccess::GlobalToGridLocal2DimDelegate gToZphiT2;
+  gToZphiT2.connect<&GlobalToZPhi::g2ZPhi>(std::move(globalToGridT2));
+
+  Acts::ProtoAxis pAxisZ(Acts::AxisDirection::AxisZ,
+                         Acts::AxisBoundaryType::Bound, -1.0, 1.0, 2);
+  Acts::ProtoAxis pAxisPhi(Acts::AxisDirection::AxisPhi,
+                           Acts::AxisBoundaryType::Closed, -std::numbers::pi,
+                           std::numbers::pi, 4);
+
+  std::vector<std::vector<std::size_t>> indexPayload = {
+      std::vector<std::size_t>{1u, 1u, 0u, 2u},
+      std::vector<std::size_t>{0u, 3u, 3u, 0u}};
+
+  auto ismZPhi = Acts::createGridSurfaceMaterial<std::size_t>(
+      pAxisZ, pAxisPhi, Acts::IndexedMaterialAccessor{std::move(materialT2)},
+      std::move(bToZPhiT2), std::move(gToZphiT2), indexPayload);
```

Comment:
The full construction is done in 10 lines of code, particularly for the indexed version this is very simple.